pkg/compact/compact.go

```diff
@@ -645,11 +703,6 @@ func (cg *Group) compact(ctx context.Context, dir string, comp tsdb.Compactor) (
 			return false, ulid.ULID{}, errors.Wrapf(err, "read meta from %s", pdir)
 		}
 
-		cgKey, groupKey := cg.Key(), GroupKey(meta.Thanos)
```

Comment:
I've removed this circuit breaker because, from my perspective, it shouldn't be the single group responsabilty to know how blocks should be grouped together (that's Grouper responsability).

Reply:
Makes sense to me (: